### ISSUE TITLE
AUTHENTICATE arguments are not split.

### ISSUE THREAD

**Issue by ProgVal** [sopel-irc/sopel#975]
When using a very long password, the authstring given with AUTHENTICATE exceeds 400 characters, and thus should be splitted, according to [the specification](http://ircv3.net/specs/extensions/sasl-3.1.html#the-authenticate-command)

Config file:

```
[core]
nick = Sopel
host = 0.0.0.0
use_ssl = false
port = 34827
owner = me
channels = 
auth_username = foo
auth_password = barbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbarbar
auth_method = sasl
```

Sopel session log:

```
C: CAP LS 302
S: CAP * LS :sasl
C: NICK Sopel
C: USER sopel +iw Sopel :Sopel: http://sopel.chat
C: CAP REQ multi-prefix
S: CAP Sopel NAK :multi-prefix
C: CAP REQ sasl
S: CAP Sopel ACK :sasl
C: AUTHENTICATE PLAIN
S: AUTHENTICATE +
C: AUTHENTICATE Zm9vAGZvbwBiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiY
```

Expected log:

```
C: CAP LS 302
S: CAP * LS :sasl
C: NICK Sopel
C: USER sopel +iw Sopel :Sopel: http://sopel.chat
C: CAP REQ multi-prefix
S: CAP Sopel NAK :multi-prefix
C: CAP REQ sasl
S: CAP Sopel ACK :sasl
C: AUTHENTICATE PLAIN
S: AUTHENTICATE +
C: AUTHENTICATE :Zm9vAGZvbwBiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJi
C: AUTHENTICATE :YXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJiYXJi
C: AUTHENTICATE :YXJiYXJiYXI=
```

Version: 047a99ed2d3f76c7d54e5bacaac1c1d38d1b743c (Git master at the time I am writing this)

**Comment by Exirel** [comment]
Since I'm working on PR #1448 to manage multi-line messages, I think I can reuse some of the logic (if not the code itself) to handle this case. I'll see what I can do, and for now I just leave a note here.

**Comment by Exirel** [comment]
Just a quick note for myself, here is an extract from the specification to implement:

> The response is encoded in Base64 (RFC 4648), then split to 400-byte chunks, and each chunk is sent as a separate AUTHENTICATE command. Empty (zero-length) responses are sent as AUTHENTICATE +. If the last chunk was exactly 400 bytes long, it must also be followed by AUTHENTICATE + to signal end of response.

I'll work on that.

**Comment by dgw** [comment]
Fixed by #1457.
